### PR TITLE
Refactor transaction and signing sagas to main process

### PR DESCRIPTION
**Pull request by Mrtenz**

### CODE REVIEW

File: src/common/store/signing.slice.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { fPrivateKey, fTxRequest } from '@fixtures';
+import type { SerializedWallet } from '@types';
+import { WalletType } from '@types';
+import { makeTx } from '@utils';
+
+import slice, { sign, signFailed, signSuccess } from './signing.slice';
+
+jest.mock('@bridge', () => ({
+  ipcBridgeRenderer: {
+    crypto: { invoke: jest.fn() },
```
Comment (FrederikBolding) [inline src/common/store/signing.slice.test.ts:11]: This mock shouldn't be needed

File: src/app/screens/SignTransaction/SignTransactionKeystore.tsx [hunk 2]
```diff
@@ -2,11 +2,12 @@ import { Body, Heading } from '@mycrypto/ui';
 import { useEffect } from 'react';
 
 import { Account, ScrollableContainer, SignBottom } from '@app/components';
-import { getSigningError, useSelector } from '@app/store';
+import { useSelector } from '@app/store';
 import { translateRaw } from '@common/translate';
 import type { SignTransactionProps } from '@types';
 import { WalletType } from '@types';
 
+import { getSigningError } from '../../../common/store';
```
Comment (FrederikBolding) [inline src/app/screens/SignTransaction/SignTransactionKeystore.tsx:10]: Use an absolute import here

File: src/api/store/signing.sagas.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  tx: TransactionRequest;
+}>) {
+  try {
+    const signedTx: string = yield call(signTransaction, wallet, tx);
+
+    yield put(signSuccess());
+
+    const currentTx = yield select(getCurrentTransaction);
+
+    ipcMain.emit(IPC_CHANNELS.API, { id: currentTx.id, result: signedTx });
```
Comment (FrederikBolding) [inline src/api/store/signing.sagas.ts:40]: Leave a comment to refactor this

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Comment by Mrtenz** [comment]
[ch8498]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**